src/components/ResultFavoritesButton/ResultFavoritesButton.tsx

```diff
@@ -8,14 +8,22 @@ interface ResultFavoritesButtonProps {
 }
 
 export default function ResultFavoritesButton(props: ResultFavoritesButtonProps) {
+  const [isFavorited, setIsFavorited] = useState(false);
```

Comment:
Lets not have any logic in presentational components and have the logic in hooks